src/embedding.rs

```diff
@@ -191,11 +201,24 @@ where
                     let value = col.get(i as usize).unwrap();
                     embedding.insert(j, *value);
                 }
-                embedding_persistor.put_data(entity_name, hash_occur, embedding);
+                embedding_persistor
+                    .put_data(&entity_name, hash_occur, embedding)
+                    .unwrap_or_else(|_| {
+                        broken_entities.insert(entity_name);
+                    });
             };
         }
 
-        embedding_persistor.finish();
+        if !broken_entities.is_empty() {
+            warn!(
+                "Entities which can't be written to the file: {:?}",
```

Comment:
Maybe display total count of broken entities and just a few first? Not sure what if it makes sense to display all of them if there is a lot of broken

Reply:
I will do it. If there are more errors probably there is a bigger issue.